### PR TITLE
fix: Generate unique custom property name

### PR DESCRIPTION
**Pull request by fralongo**
Currently, the hash of a custom CSS property for a token depends only on the resolved values of token in the primary theme. It does not depend on
- the resolved values of the token in contexts
- the resolved values of the token in secondary themes and contexts.

This creates conflicts when an application loads Cloudscape components with different themes.
This PR calculates the hash of a custom property based on all the resolved values for a token across themes and contexts.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/build/__tests__/properties.test.ts [hunk 1]
```diff
@@ -1,10 +1,164 @@
...
+        tokens: { shadow: { light: 'green', dark: 'blue' } },
+        contexts: { navigation: { ...navigationContext, tokens: { shadow: 'black' } } },
+      },
+    ],
+    preset.variablesMap
+  );
+  expect(firstMap.shadow).not.toEqual(secondMap.shadow);
+});
+
+test.only('generates same hash if context defines another token', () => {
```
Comment (rubencarvalho) [inline src/build/__tests__/properties.test.ts:60]:
```suggestion
test('generates same hash if context defines another token', () => {
```

File: src/build/properties.ts [hunk 2]
```diff
@@ -1,22 +1,40 @@
...
+
+  const mapEntries = Object.entries(tokensValuesMap).map(([token, values]) => {
     return [token, toCssVarName(variablesMap[token], values.filter((value) => !!value) as Array<string>)];
   });
   return Object.fromEntries(mapEntries);
 }
+
+const getTokensValuesFromResolutions = (resolutions: Array<FullResolution>): TokensValuesMap =>
+  resolutions.reduce((tokensValuesMap: TokensValuesMap, resolution: FullResolution) => {
+    Object.entries(resolution).map(([token, assignment]) => {
```
Comment (rubencarvalho) [inline src/build/properties.ts:32]: nit: this `.map` tricked me a bit while reading the function - it seems we're not returning anything on the loop. Maybe we could have a `.forEach` instead?

File: src/build/properties.ts [hunk 3]
```diff
@@ -1,22 +1,40 @@
...
+      const values = typeof assignment === 'string' ? [assignment] : Object.values(assignment);
+      if (!tokensValuesMap[token]) {
+        tokensValuesMap[token] = [];
+      }
+      tokensValuesMap[token] = [...tokensValuesMap[token], ...values];
```
Comment (rubencarvalho) [inline src/build/properties.ts:37]: Here we're modifying the `tokensValuesMap` directly. I'm not very familiar with the package, but is there a risk of an unwanted side effect if this is used somewhere else?

Alternatively, we could create a new object, modify and return it instead.
```typescript
const newTokensValuesMap = { ...tokensValuesMap };
```

**Review by rubencarvalho — COMMENTED** [review]
Left a couple of comments, everything else looks good to me! The tests made it clear what is happening :)

File: src/build/properties.ts (hunk 3 above)
Comment (fralongo) [reply src/build/properties.ts:37]: The `tokensValuesMap ` is initialized in the reduce loop within this function, so it is not used elsewhere.

Comment (rubencarvalho) [reply src/build/properties.ts:37]: Ah, yes. I was looking at this function in isolation. LGTM!

**Review by rubencarvalho — APPROVED** [review]

**Pull request merged**